src/installer/records.py

```diff
@@ -111,6 +123,22 @@ def __repr__(self):
             self.path, self.hash_, self.size,
         )
 
+    def __eq__(self, other):
+        # type: (object) -> bool
+        if not isinstance(other, Record):
+            return NotImplemented
+        return (
+            self.path == other.path
```

Comment:
Isn't the relevant part here the path? I feel like if we have multiple records with the same path but different hashes/size there is probably something wrong, I would raise and exception.

Reply:
I don't think it's a good idea to raise an exception while checking for equality. I would find it surprising as a user. As an equality check I don't think we should make assumptions about how it will be used. Consider a case where we rewrite the shebang in a script: if one builds a record of the original file and one for the installed file, the records will have different hashes and likely sizes as well. These should just compare as not equal in my opinion.

Reply:
I appreciate the semantics we get from just having this return `False`. As you explain it does make sense from a logical/structural POV. But I also think the benefits of having this explode on unwanted cases heavily outweight that.

I feel like the current API induces users to wrongly use it. But I guess it comes down to how you promote it being used.
In which use cases would you use this? Or rather, which problem does this solve?
(I wait for your response to proceed the discussion in order to avoid bikesheding :sweat_smile:)